cartographer_grpc/framework/server.h

```diff
@@ -57,10 +58,11 @@ class Server {
           RpcHandlerInfo{
               RpcHandlerType::RequestType::default_instance().GetDescriptor(),
               RpcHandlerType::ResponseType::default_instance().GetDescriptor(),
-              [](Rpc* const rpc) {
+              [](Rpc* const rpc, ExecutionContext* const execution_context) {
```

Comment:
This factory function looks pretty standard. Is it going to have more logic in the future?
If not, can it not be created here but the simple logic just be executed where it is needed?

Reply:
We need the type information to instantiate the concrete handler type. So somehow that type information needs to get to the place where the handler is instantiated, i.e. Rpc. Here we "capture" the type information in the closure. An alternative would be to have templated `RpcImpl<HandlerType>` that implement Rpc. But that would only move the problem to Rpc which then also have to be created using some factory.